src/OrthoTree/OrthoTree.h

```diff
@@ -1,12 +1,167 @@
+        const size_t max_depth_m;
+        const size_t max_particles_per_node_m;
+        const bounds_t root_bounds_m;
+        const Morton<Dim> morton_helper;
+
+        // as of now the tree is stored only as morton codes, this needs to be discussed as a group
+        Kokkos::vector<morton_code> tree_m;
+
+        size_t n_particles;
+
+        // this list should not be edited, we will probably need it if we implement the tree update as well
+        aid_list_t aid_list;
+
+    public:
+
+        OrthoTree(size_t max_depth, size_t max_particles_per_node, const bounds_t& root_bounds);
+
+        /**
+         * @brief This is the most basic way to build a tree. Its inefficien, but it (should) be correct.
+         * Can be used to compare against parallel implementations later on.
+         *
+         * @param particles An 'object of arrays' of particles (google it)
+         */
+        void build_tree_naive(particle_t const& particles);
+
+        /**
+         * @brief Compares the following aspects of the trees:
+         * - n_particles
+         * - tree_m.size()
+         * - contents of tree_m
+         *
+         * @warning THIS FUNCTION ASSUMES THAT THE TREES ARE SORTED
+         *
+         * @param other
+         * @return true
+         * @return false
+         */
+        bool operator==(const OrthoTree& other);
+
+        /**
+         * @brief Returns a vector with morton_codes and lists of particle ids inside this region
+         * This is also not meant to be permanent, but it should suffice for the beginning
+         * @return Kokkos::vector<Kokkos::pair<morton_code, Kokkos::vector<size_t>>>
+         */
+        Kokkos::vector<Kokkos::pair<morton_code, Kokkos::vector<size_t>>> get_tree() const;
+
+    private:
+
+        /**
+         * @brief initializes the aid list which has form vec{pair{morton_code, size_t}}
+         * will sort the aidlist in ascending morton codes
+         *
+         * @param particles
+         */
+        void initialize_aid_list(particle_t const& particles);
+
+    public:
+
+        // SIMONS FUNCTIONS DONT EDIT, TOUCH OR USE THIS IN YOUR CODE:
+
+        /**
+         * @brief algorithm 1' topdown sequential construction of octree
+         *
+         * @param morton code of root_node, particles
+         *
+         * @return list of morton codes of leaves of tree spanning root node
+         **/
+        ippl::vector_t<morton_code> build_tree_topdown_sequential(morton_code root_node, particle_t const& particles)
+        {
+            // insert the root into the tree
+            ippl::vector_t<morton_code> tree;
+            ippl::vector_t<morton_code> stack; // stack used to build the tree
+
+            stack.push_back(root_node); // maybe store morton_code(0) s.t. we can call morton::root_val or smth
+
+            initialize_aid_list(particles); // initialize aid list - required for particle counting could be moved to constructor
+
+            while ( stack.size() > 0 ) {
```

Comment:
This traversal should be possible without a stack.
It's an open question whether this is worth it as stack size should be proportional to max_depth here

Reply:
i will leave this as it is for now. not relevant for mvp.